### PR TITLE
[PENDING API REVIEW] add list broadcasts api

### PR DESCRIPTION
**Pull request by msach22**
#### What is this PR doing?
Adds List Broadcasts API

#### How should this be manually tested?

```javascript
var options = {
  sessionId: 'SESSIONID', // sessionId that you want to filter for (optional)
  offset: '', // optional
  count: '', // optional
}; 
opentok.listBroadcasts(options, function(error, broadcastList, totalCount) {
  if (error) {
   console.log('error', error);
  } else {
   console.log('broadcastList', broadcastList);
   console.log('totalCount', totalCount);
  } 
});
```
#### What are the relevant tickets?
Internal tickets

### CODE REVIEW

File: test/opentok-test.js [hunk 1]
```diff
@@ -51,6 +51,22 @@ var mockBroadcastObject = {
...
+    maxDuration: 7200,
+    resolution: '1280x720',
+    event: 'broadcast',
+    status: 'stopped'
+  }]
+};
+
 function validateBroadcastObject(broadcast, status) {
   expect(broadcast.id).to.equal('fooId');
   expect(broadcast.id).to.equal('fooId');
```
Comment (jeffswartz) [inline test/opentok-test.js]: You can remove this line.

File: test/opentok-test.js [hunk 2]
```diff
@@ -1389,6 +1431,59 @@ describe('#getBroadcast', function () {
...
+      expect(err).to.be.null;
+      validateListBroadcastsObject(broadcastList);
+      validateTotalCount(totalCount);
+      done();
+    });
+  });
+
+  it('succeeds given options as valid parameters', function (done) {
+    mockListBroadcastsRequest({
+      sessionId: 'SESSIONID',
```
Comment (jeffswartz) [inline test/opentok-test.js]: I get a comma-dangle lint error here. There are also a number of other lint errors (some of which exist on the master branch, but they are easy fixes).

File: lib/client.js [hunk 3]
```diff
@@ -167,6 +169,26 @@ Client.prototype.getBroadcast = function getBroadcast(broadcastId, cb) {
   });
 };
 
+Client.prototype.listBroadcasts = function listBroadcasts(queryString, cb) {
+  var baseUrl = this.c.apiUrl + this.c.endpoints.listBroadcasts.replace(/<%apiKey%>/g, this.c.apiKey);
+  var url = queryString.length > 0 ? baseUrl + '?' + queryString.join('&') : baseUrl;
```
Comment (jeffswartz) [inline lib/client.js]: It kind of bugs me that `queryString` is an array. Can we call `join()` at the end of the `listBroadcasts()` method in opentok.js?

Comment (msach22) [reply lib/client.js]: I was just keeping it similar to the listArchives function, but happy to do that in the `opentok.listBroadcasts` method and pass in a string instead.

File: test/opentok-test.js (hunk 1 above)
Comment (msach22) [reply test/opentok-test.js]: Hmm, I'm just leaving a line between the next function. I think it improves readability and it's how all of the other functions are written.

File: test/opentok-test.js (hunk 2 above)
Comment (msach22) [reply test/opentok-test.js]: I fixed the linter issues for the list broadcasts changes. Happy to do the others in a separate PR.

File: lib/client.js (hunk 3 above)
Comment (msach22) [reply lib/client.js]: Fixed

File: test/opentok-test.js (hunk 1 above)
Comment (jeffswartz) [reply test/opentok-test.js]: The following line is included twice: `expect(broadcast.id).to.equal('fooId');`. We can remove one of them.

Comment (msach22) [reply test/opentok-test.js]: Ah great catch - fixed!

**Pull request merged**